kinder/pkg/kindconfig/validate.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kindconfig
+
+import (
+	"net"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/kind/pkg/cluster/config/v1alpha3"
+	"sigs.k8s.io/kind/pkg/util"
+)
+
+// This file contains validation for the kind Cluster and Node types copied from the
```

Comment:
I think this code should go under third_party folder, so it is more explicit this is a fork we should take care to re-check every time we upgrade the kind dependency.
WDYT?

Reply:
ok, it however means that third_party needs to import our local Cluster object.
and it can no longer be `func (c *Cluster) Validate()` but instead `ValidateCluster(....)`

are you OK with that?

Reply:
What about `ValidateCluster(cluster v1alpha3.Cluster)` so there is no need of importing our local cluster object?

Reply:
it can work.